### PR TITLE
OSDOCS-5152 and 5158 Add support for Shielded and Confidential VMs

### PR DESCRIPTION
**Pull request by bscott-rh**
Version(s):
4.13

Issue:
https://issues.redhat.com/browse/OSDOCS-5152
https://issues.redhat.com/browse/OSDOCS-5158

Link to docs preview:
[Installation configuration parameters table](https://56805--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-customizations.html#installation-configuration-parameters-additional-gcp_installing-gcp-customizations)
[Enabling Shielded VMs](https://56805--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-customizations.html#installation-gcp-enabling-shielded-vms_installing-gcp-customizations)
[Enabling Confidential VMs](https://56805--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-customizations.html#installation-gcp-enabling-confidential-vms_installing-gcp-customizations)

The content is the same in all of the GCP assemblies, so I only included preview links for one assembly.

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by bscott-rh** [comment]
@jianli-wei hello Jianli, PTAL at this PR for shielded and confidential VM documentation on GCP. Thanks

File: modules/installation-gcp-enabling-confidential-vms.adoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// * installing/installing_gcp/installing-gcp-vpc.adoc
+// * installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.adoc
+// * installing/installing_gcp/installing-gcp-user-infra.adoc
+// * installing/installing_gcp/installing-gcp-user-infra-vpc.adoc
+// * installing/installing_gcp/installing-restricted-networks-gcp.adoc
+
+:_content-type: PROCEDURE
+[id="installation-gcp-enabling-confidential-vms_{context}"]
+= Enabling Confidential VMs
+In {product-title} {product-version}, you can use Confidential VMs when installing your cluster. Shielded VMs encrypt data while it is being processed. For more information, see Google's documentation on link:https://cloud.google.com/confidential-computing[Confidential Computing]. You can enable Confidential VMs and Shielded VMs at the same time, although they are not dependent on each other.
```
Comment (jianli-wei) [inline modules/installation-gcp-enabling-confidential-vms.adoc:15]: It seems `Shielded VMs encrypt data while it is being processed` should be `Confidential VMs encrypt data while it is being processed`, thanks!

**Review by jianli-wei — CHANGES_REQUESTED** [review]

File: modules/installation-gcp-enabling-confidential-vms.adoc (hunk 1 above)
Comment (bscott-rh) [reply modules/installation-gcp-enabling-confidential-vms.adoc:15]: Great catch, thanks

File: modules/installation-configuration-parameters.adoc [hunk 2]
```diff
@@ -1246,6 +1246,18 @@ link:https://yaml.org/spec/1.2/spec.html#sequence//[YAML sequence].
...
 |The GCP service account email, for example `<service_account_name>@<project_id>.iam.gserviceaccount.com`.
 
+|`platform.gcp.defaultMachinePlatform.secureBoot`
+|Whether to enable Shielded VMs for all machines in the cluster. Shielded VMs have additional security protocols such as secure boot, firmware and integrity monitoring, and rootkit protection. For more information on Shielded VMs, see Google's documentation on link:https://cloud.google.com/shielded-vm[Shielded VMs].
+|`Enabled` or `Disabled`. The default value is `Disabled`.
+
+|`platform.gcp.defaultMachinePlatform.confidentialCompute`
+|Whether to enable Confidential VMs for all machines in the cluster. Confidential VMs provide encryption for data while it is being processed. For more information on Confidential VMs, see Google's documentation on link:https://cloud.google.com/confidential-computing[Confidential Computing].
```
Comment (eranco74) [inline modules/installation-configuration-parameters.adoc:1254]: We can say `Enable confidential computing` or `use confidential VMs` , I'm not sure about  `enable confidential VMs`
```suggestion
|Whether to use Confidential VMs for all machines in the cluster. Confidential VMs provides encryption for data while it is being processed. For more information on Confidential computing, see Google's documentation on link:https://cloud.google.com/confidential-computing[Confidential Computing].
```

File: modules/installation-configuration-parameters.adoc [hunk 3]
```diff
@@ -1246,6 +1246,18 @@ link:https://yaml.org/spec/1.2/spec.html#sequence//[YAML sequence].
 |The GCP service account used for the encryption request for control plane and compute machines. If absent, the Compute Engine default service account is used. For more information about GCP service accounts, see Google's documentation on link:https://cloud.google.com/compute/docs/access/service-accounts#compute_engine_service_account[service accounts].
 |The GCP service account email, for example `<service_account_name>@<project_id>.iam.gserviceaccount.com`.
 
+|`platform.gcp.defaultMachinePlatform.secureBoot`
+|Whether to enable Shielded VMs for all machines in the cluster. Shielded VMs have additional security protocols such as secure boot, firmware and integrity monitoring, and rootkit protection. For more information on Shielded VMs, see Google's documentation on link:https://cloud.google.com/shielded-vm[Shielded VMs].
```
Comment (eranco74) [inline modules/installation-configuration-parameters.adoc:1250]: This option is specific to secure boot which is just one of the shielded VMs options. 
Consider:
```suggestion
|Whether to enable Shielded VMs secure boot for all machines in the cluster. Shielded VMs have additional security protocols such as secure boot, firmware and integrity monitoring, and rootkit protection. For more information on Shielded VMs, see Google's documentation on link:https://cloud.google.com/shielded-vm[Shielded VMs].
```

File: modules/installation-gcp-enabling-confidential-vms.adoc [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+----
+controlPlane:
+  platform:
+    gcp:
+       confidentialCompute: Enabled <1>
+       type: n2d-standard-8 <2>
+       onHostMaintenance: Terminate <3>
+----
+<1> Enable confidential VMs. 
+<2> Specify a machine type that supports Confidential VMs. Confidential VMs require the N2D or C2D series of machine types.
```
Comment (eranco74) [inline modules/installation-gcp-enabling-confidential-vms.adoc:35]: You can refer to this link
https://cloud.google.com/compute/confidential-vm/docs/os-and-machine-type#machine-type

**Review by eranco74 — COMMENTED** [review]
Looks great

**Comment by bscott-rh** [comment]
@jianli-wei Thank you for reviewing the PR. any additional feedback before I move forward to peer review?

**Comment by eohartman** [comment]
@bscott-rh This peer review might spill into Tuesday, as I am newer to the peer review squad and still need a bit of hand-holding, though I've been encouraged to take on bigger items. If the timing is an issue, let me know and I'll let someone else take this.

File: modules/installation-gcp-enabling-confidential-vms.adoc [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+// * installing/installing_gcp/installing-gcp-vpc.adoc
+// * installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.adoc
+// * installing/installing_gcp/installing-gcp-user-infra.adoc
+// * installing/installing_gcp/installing-gcp-user-infra-vpc.adoc
+// * installing/installing_gcp/installing-restricted-networks-gcp.adoc
+
+:_content-type: PROCEDURE
+[id="installation-gcp-enabling-confidential-vms_{context}"]
+= Enabling Confidential VMs
+In {product-title} {product-version}, you can use Confidential VMs when installing your cluster. Confidential VMs encrypt data while it is being processed. For more information, see Google's documentation on link:https://cloud.google.com/confidential-computing[Confidential Computing]. You can enable Confidential VMs and Shielded VMs at the same time, although they are not dependent on each other.
```
Comment (mjpytlak) [inline modules/installation-gcp-enabling-confidential-vms.adoc:15]: @bscott-rh In both modules, I suggest getting rid of `In {product-title} {product-version},`

**Comment by mjpytlak** [comment]
@bscott-rh Based on https://issues.redhat.com/browse/OCPBU-142, I believe confidential VMs is shipping in Tech Preview and the content requires require the TP snippet. Further, please work with @JoeAldinger to make sure that confidential computing is listed as TP in the release notes.

CC @jeana-redhat as I believe you also have doc for this. Want to make sure you are aware of Tech Preview.

**Comment by bscott-rh** [comment]
> @bscott-rh Ready to merge at this point?

Ready to go, I added the TP snippet in the confidential module. I will update the RNs next.

**Comment by JoeAldinger** [comment]
> > @bscott-rh Ready to merge at this point?
> 
> Ready to go, I added the TP snippet in the confidential module. I will update the RNs next.

cool, cool. Thanks for looping me in @mjpytlak and @bscott-rh it seems like all is good?

File: modules/installation-gcp-enabling-confidential-vms.adoc [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+// * installing/installing_gcp/installing-gcp-user-infra-vpc.adoc
+// * installing/installing_gcp/installing-restricted-networks-gcp.adoc
+
+:_content-type: PROCEDURE
+[id="installation-gcp-enabling-confidential-vms_{context}"]
+= Enabling Confidential VMs
+
+You can use Confidential VMs when installing your cluster. Confidential VMs encrypt data while it is being processed. For more information, see Google's documentation on link:https://cloud.google.com/confidential-computing[Confidential Computing]. You can enable Confidential VMs and Shielded VMs at the same time, although they are not dependent on each other.
+
+:FeatureName: Confidential Computing
```
[the remark this answers is not in the document]
Comment (jeana-redhat) [reply modules/installation-gcp-enabling-confidential-vms.adoc:18]: #57925

**Comment by mjpytlak** [comment]
> it seems like all is good?

@JoeAldinger Not sure. Does @bscott-rh update the Tech Preview table or does the RN team?

**Pull request merged**